### PR TITLE
Unlock remove

### PR DESCRIPTION
**Pull request by josegomez**
This Adds functionality to Remove SQRL account as well as the unlock functionality.

Disable / Enable Demo:
DisableReEnable

Remove Demo:
RemoveAccount

This is ready to Merge I think once @alexhauser gives it a once over.

Added the following strings which may need some correction on the German Side (Google Translate was used in the meantime)
ReEnableSQRLTitle
SqrlDisabledAlert
InvalidRescueCodeMessage

### CODE REVIEW

**Comment by josegomez** [comment]
Also @alexhauser  there appears to be a little bug in the InputSecretDialog (for Rescue Code) where when the dialog pops up you can't type in it right away
It works fine if you fiddle around with the form maximize, change size move it around... I can't quite figure out why any ideas there?

File: SQRLDotNetClientUI/ViewModels/AuthenticationViewModel.cs [hunk 1]
```diff
@@ -200,6 +201,49 @@ public async void Login()
...
+                                    serverResponse = SQRL.GenerateEnableCommand(serverResponse.NewNutURL, siteKvp, serverResponse.FullServerRequest, ursKey, addClientData, sqrlOpts);
+                                }
+                                else
+                                {
+                                    var msgBox = MessageBox.Avalonia.MessageBoxManager.GetMessageBoxStandardWindow(
+                                        _loc.GetLocalizationValue("ErrorTitleGeneric"),
+                                        _loc.GetLocalizationValue("InvalidRescueCodeMessage"),
+                                        MessageBox.Avalonia.Enums.ButtonEnum.YesNo,
+                                        MessageBox.Avalonia.Enums.Icon.Error);
+                                    var answer =await msgBox.ShowDialog(AvaloniaLocator.Current.GetService<MainWindow>());
```
Comment (alexhauser) [inline SQRLDotNetClientUI/ViewModels/AuthenticationViewModel.cs]: I've added a field named `_mainWindow` to `ViewModelBase` which is available to all the view models that inherit from it. Makes it so much easier to read.

File: SQRLDotNetClientUI/ViewModels/AuthenticationViewModel.cs [hunk 2]
```diff
@@ -243,7 +287,33 @@ public async void Login()
...
+                                        _mainWindow.Close();
+                                    }
+                                    else
+                                    {
+                                        var msgBox = MessageBox.Avalonia.MessageBoxManager.GetMessageBoxStandardWindow(
+                                        _loc.GetLocalizationValue("ErrorTitleGeneric"),
+                                        _loc.GetLocalizationValue("InvalidRescueCodeMessage"),
+                                        MessageBox.Avalonia.Enums.ButtonEnum.Ok,
+                                        MessageBox.Avalonia.Enums.Icon.Error);
+                                        await msgBox.ShowDialog(AvaloniaLocator.Current.GetService<MainWindow>());
```
Comment (alexhauser) [inline SQRLDotNetClientUI/ViewModels/AuthenticationViewModel.cs]: Same here, you can use `_mainWindow`.

File: SQRLDotNetClientUI/ViewModels/AuthenticationViewModel.cs [hunk 3]
```diff
@@ -200,6 +201,49 @@ public async void Login()
...
+                                Icon.Lock);
+                            messageBoxStandardWindow.SetMessageStartupLocation(Avalonia.Controls.WindowStartupLocation.CenterOwner);
+                            var btResult = await messageBoxStandardWindow.ShowDialog(_mainWindow);
+                            if (btResult == ButtonResult.Yes)
+                            {
+                                RetryRescueCode:
+                                InputSecretDialogView rescueCodeDlg = new InputSecretDialogView(SecretType.RescueCode);
+                                rescueCodeDlg.WindowStartupLocation = WindowStartupLocation.CenterOwner;
+                                string rescueCode = await rescueCodeDlg.ShowDialog<string>(
+                                    AvaloniaLocator.Current.GetService<MainWindow>());
```
Comment (alexhauser) [inline SQRLDotNetClientUI/ViewModels/AuthenticationViewModel.cs]: See my other comments about `_mainWindow`.

File: SQRLDotNetClientUI/Assets/Localization/localization.json [hunk 4]
```diff
@@ -285,7 +294,7 @@
     { "EncryptingBlock2Label": "Verschlüssle Block 2" },
     { "BtnNext": "Weiter" },
     { "InvalidPasswordMessage": "Ungültiges Passwort!" },
-    { "InvalidRescueCodeMessage": "Ungültiger Rettungscode (\"rescue code\")!" },
+    { "InvalidRescueCodeMessage": "Ungültiger Rettungscode (\"rescue code\")! Bitte versuche es erneut" },
```
Comment (alexhauser) [inline SQRLDotNetClientUI/Assets/Localization/localization.json]: `{ "InvalidRescueCodeMessage": "Ungültiger Rettungscode (\"rescue code\")! Bitte versuchen Sie es erneut." },`

**Review by alexhauser — APPROVED** [review]
Great job, Jose! And Google translate is getting better as well ;-)
I'll review it a bit more tomorrow once it is merged. Heading to bed now.

**Comment by alexhauser** [comment]
> Also @alexhauser there appears to be a little bug in the InputSecretDialog (for Rescue Code) where when the dialog pops up you can't type in it right away
> It works fine if you fiddle around with the form maximize, change size move it around... I can't quite figure out why any ideas there?

Going to look at this tomorrow, @josegomez!

**Pull request merged**